mod/ArchConsoleManager.cs

```diff
@@ -0,0 +1,189 @@
+
+        private void Start()
+        {
+            GlobalMessenger.AddListener("EnterConversation", () => gameplayConsole.SetActive(false));
+            GlobalMessenger.AddListener("ExitConversation", () => gameplayConsole.SetActive(true));
+        }
+
+        private void Update()
+        {
+            // Show the correct version of the console depending on if the game is paused or not
+            if (isPaused != Randomizer.Instance.ModHelper.Menus.PauseMenu.IsOpen)
+            {
+                isPaused = !isPaused;
+                ShowConsoles(isPaused);
+            }
+        }
+
+        // Creates the two console displays
+        private void CreateConsoles(OWScene scene, OWScene loadScene)
+        {
+            if (loadScene != OWScene.SolarSystem && loadScene != OWScene.EyeOfTheUniverse) return;
+            // Create objects and establish references
+            pauseConsoleEntries = new List<GameObject>();
+            gameplayConsoleEntries = new List<GameObject>();
+            pauseConsole = GameObject.Instantiate(Randomizer.Assets.LoadAsset<GameObject>("ArchRandoCanvas"));
+            pauseConsoleContent = pauseConsole.transform.Find("PauseConsole/Scroll View/Viewport/Content").gameObject;
+            pauseConsoleViewer = pauseConsole.transform.GetChild(0).gameObject;
+            gameplayConsole = pauseConsole.transform.Find("GameplayConsole").gameObject;
+            foreach (Transform child in gameplayConsole.transform)
+            {
+                gameplayConsoleEntries.Add(child.gameObject);
+                child.GetComponent<Text>().text = string.Empty;
+            }
+            pauseConsoleContent.AddComponent<FixConsoleLayout>();
+            gameplayConsole.AddComponent<FixConsoleLayout>();
+
+            // Copy text over from previous loops
+            foreach (string entry in consoleMasterList)
+            {
+                AddText(entry, true, true);
+            }
+            pauseConsole.GetComponentInChildren<InputField>().onEndEdit.AddListener(OnConsoleEntry);
+            consoleText = pauseConsole.GetComponentInChildren<InputField>();
+            pauseConsoleViewer.SetActive(false);
+
+            StartCoroutine(LoopGreeting());
+        }
+
+        // Shows the appropriate consoles when the game is paused or not
+        private void ShowConsoles(bool showPauseConsole)
+        {
+            pauseConsoleViewer.SetActive(showPauseConsole);
+            gameplayConsole.SetActive(!showPauseConsole);
+        }
+
+        /// <summary>
+        /// Adds a new text entry to the in-game consoles
+        /// </summary>
+        /// <param name="text">The text to add to the consoles</param>
+        /// <param name="skipGameplayConsole">Whether to only show text on the pause console</param>
+        /// <param name="skipMasterList">Whether to not save this text between loops</param>
+        public void AddText(string text, bool skipGameplayConsole = false, bool skipMasterList = false)
```

Comment:
2nd argument is intended to be used to prevent text added to the console from being shown in the gameplay console, and only show in the pause menu console. This makes it look a bit cleaner, since you can filter out certain kinds of messages that don't need to be shown during gameplay (such as technical messages). I use this when reapplying text from previous loops, so that you have a clear log when starting a new loop (but can still view the pause screen for messages from previous loops).
3rd argument is mostly just used so that when I'm reapplying text from previous loops, I don't modify the Master List of text entries that I'm simultaneously reading from in order to copy the text over. This can also be used just to make sure some text doesn't save between loops if it's some really minor thing, if desired.

Reply:
Ah, keeping full history in only the pause console is a good use case for skipGameplayConsole, that makes sense now. All the confusion around the 3rd argument made that unclear.

Still, why not split AddText() into two methods?

Reply:
I don't see the purpose of splitting it. In most cases you would want skipGameplayConsole to be false anyway, so putting all of that in a separate method would mean calling two functions instead of one in most cases. Unless you mean call the second method from within this one, in which case I still don't see the point for such a small amount of code.